internal/controller/instaslice_daemonset.go

```diff
@@ -185,130 +175,91 @@ func (r *InstaSliceDaemonsetReconciler) Reconcile(ctx context.Context, req ctrl.
+					if migDevice.start == allocations.Start && migDevice.uuid == allocations.GPUUUID {
+						if errCreatingConfigMap := r.createConfigMap(ctx, migUuid, existingAllocations.Namespace, allocations.Resourceidentifier); errCreatingConfigMap != nil {
+							return ctrl.Result{RequeueAfter: requeue1sDelay}, nil
 						}
 
-						giInfo, retForGiInfor := gi.GetInfo()
-						if retForGiInfor != nvml.SUCCESS {
-							log.Error(retForGiInfor, "error getting GPU instance info for ", "giInfo", &giInfo)
-
-						}
-						//TODO: figure out the compute slice scenario, I think Kubernetes does not support this use case yet
-						ciProfileInfo, retCodeForCiProfile := gi.GetComputeInstanceProfileInfo(ciProfileId, 0)
-						if retCodeForCiProfile != nvml.SUCCESS {
-							log.Error(retCodeForGiWithPlacement, "error getting compute instance profile info for ", "pod", allocations.PodName)
-						}
-						ci, retCodeForComputeInstance := gi.CreateComputeInstance(&ciProfileInfo)
-						if retCodeForComputeInstance != nvml.SUCCESS {
-							log.Error(retCodeForComputeInstance, "error creating Compute instance for ", "ci", ci)
+						if errAddingPrepared := r.createPreparedEntry(ctx, allocations.Profile, allocations.PodUUID, allocations.GPUUUID, migDevice.giInfo.Id, migDevice.ciInfo.Id, &instaslice, migUuid); errAddingPrepared != nil {
+							return ctrl.Result{RequeueAfter: requeue1sDelay}, nil
 						}
-						migInfos, err := populateMigDeviceInfos(device)
+
+						updateInstasliceObject, err := r.getInstasliceObject(ctx, instaslice.Name, instaslice.Namespace)
 						if err != nil {
-							log.Error(err, "unable to iterate over newly created mig devices")
+							return ctrl.Result{RequeueAfter: requeue1sDelay}, nil
+						}
+
+						updatedAllocation := updateInstasliceObject.Spec.Allocations[allocations.PodUUID]
+						// updated object is still in creating status, chances are user has not yet deleted
+						// set status to created.
+						if updatedAllocation.Allocationstatus == existingAllocations.Allocationstatus {
+							existingAllocations.Allocationstatus = inferencev1alpha1.AllocationStatusCreated
+						} else {
+							// Add the new allocation status which is not created and let the daemonset handle in next reconcile
+							log.Info("allocation status changed for ", "pod", allocations.PodName, "status", updatedAllocation.Allocationstatus)
+							existingAllocations.Allocationstatus = updatedAllocation.Allocationstatus
 						}
-						for migUuid, migDevice := range migInfos {
-							if allocations.Size == migDevice.size && allocations.Start == migDevice.start && migDevice.uuid == allocations.GPUUUID {
-								cachedPreparedMig[allocations.PodUUID] = preparedMig{gid: migDevice.giInfo.Id, miguuid: migUuid, cid: migDevice.ciInfo.Id}
-							}
+						updateInstasliceObject.Spec.Allocations[allocations.PodUUID] = existingAllocations
+						errForUpdate := r.Update(ctx, updateInstasliceObject)
+						if errForUpdate != nil {
```

Comment:
s/errForUpdate/err

Reply:
line 168 has a variable declared with name err

Reply:
why not reuse it? `err = r.Update(ctx, updateInstasliceObject)` ?